analyzers/src/SonarAnalyzer.CSharp/Rules/CheckArgumentException.cs

```diff
@@ -71,23 +72,20 @@ private static void CheckForIssue(SyntaxNodeAnalysisContext analysisContext)
                 return;
             }
 
-            var parameterNameValue = new Optional<object>();
-            var messageValue = new Optional<object>();
+            var parameterNameValue = default(Optional<object>);
+            var messageValue = default(Optional<object>);
             for (var i = 0; i < methodSymbol.Parameters.Length; i++)
             {
+                var argument = objectCreation.ArgumentList.Arguments[i];
                 var argumentExpression = objectCreation.ArgumentList.Arguments[i].Expression;
-                if (methodSymbol.Parameters[i].MetadataName == "paramName" || methodSymbol.Parameters[i].MetadataName == "parameterName")
-                {
-                    parameterNameValue = analysisContext.SemanticModel.GetConstantValue(argumentExpression);
-                }
-                else if (methodSymbol.Parameters[i].MetadataName == "message")
-                {
-                    messageValue = analysisContext.SemanticModel.GetConstantValue(argumentExpression);
-                }
-                else
-                {
-                    // do nothing
-                }
+                RetrieveParameterAndMessageArgumentValue(
+                    argument.NameColon != null
+                        ? argument.NameColon.Name.Identifier.ValueText
+                        : methodSymbol.Parameters[i].MetadataName,
+                    argumentExpression,
+                    analysisContext.SemanticModel,
+                    ref parameterNameValue,
+                    ref messageValue);
```

Comment:
the logic of these lines 75-88 seems cumbersome - at the end of the day, after this loop `parameterNameValue` and `messageValue` will have a single value 

wouldn't it be simpler to extract the logic into a method that would return a tuple with the results?

you could move the logic to `RetrieveParameterAndMessageArgumentValue`  and like this you can get rid of the `ref` arguments which are a code smell to me